src/canvas/PatternPath.tsx

```diff
@@ -94,4 +99,49 @@ export class PatternPath implements IPatternPath {
         const midPoint = this._computeMiddlePoint(prevPoint, currPoint);
         this._path2D.quadraticCurveTo(prevPoint.getX(), prevPoint.getY(), midPoint.getX(), midPoint.getY());
     }
+
+    private _squaredDistance = (point1: Point, point2: Point): number => {
+        const dx = point2.getX() - point1.getX();
+        const dy = point2.getY() - point1.getY();
+        return dx*dx + dy*dy;
+    }
+
+    // Returns a selection of points from the _points array that sum up the points
+    private _selectPoints = (): Point[] => {
+        const result = new Array<Point>();
+        if (this._points.length > 2) {
+            // Always include the first point
+            result.push(this._points[0]);
+            let lastIndexTaken = 0;
+            for(let i = 0;i < this._points.length - 1;i++) {
+                // Discard points that are too close in index or in distance to the last added point
+                if (i - lastIndexTaken > MIN_SKIPPED_POINTS && 
+                        this._squaredDistance(this._points[i], this._points[lastIndexTaken]) 
+                        > MIN_SQUARED_DISTANCE_BETWEEN_POINTS) {
+                    result.push(this._points[i]);
+                    lastIndexTaken = i;
+                }
+            }
+            // Always include the last point
+            result.push(this._points[this._points.length - 1]);
+        }
+        return result;
+    }
+
+    // Replaces the path2D of the patternPath with a smoothed path
+    public smoothPath = (): void => {
```

Comment:
We should probably add smoothPath to the IPatternPath type in types.d.ts. Also, would you please move it up under AddPoint to keep the public functions at the top of the file and the private functions below them.

Reply:
Would you please add to the comment which type of smoothing the algorithm does (based on the different ways we discussed we could do it).

Reply:
done!

Reply:
commenting got me thinking about the smoothing logic, and I decided to change a little bit how it works, let me know if that works for you.

Reply:
It looks like it works for me, what did you change?